Add APISIX metrics monitoring (#9725)
apache/skywalking: docs/en/setup/backend/backend-apisix-monitoring.md

# APISIX monitoring
## APISIX  performance from `apisix prometheus plugin`
SkyWalking leverages OpenTelemetry Collector to transfer the metrics to
[OpenTelemetry receiver](opentelemetry-receiver.md) and into the [Meter System](./../../concepts-and-designs/meter.md).

### Data flow

> 1. [APXSIX prometheus plugin](https://apisix.apache.org/docs/apisix/plugins/prometheus/) collect metrics data from APSIX.

Reviewer: ```suggestion
1. [APXSIX Prometheus plugin](https://apisix.apache.org/docs/apisix/plugins/prometheus/) collects metrics data from APSIX.
```

> 2. OpenTelemetry Collector fetches metrics from [APXSIX prometheus plugin](https://apisix.apache.org/docs/apisix/plugins/prometheus/) via Prometheus Receiver and pushes metrics to SkyWalking OAP Server via the OpenCensus gRPC Exporter or OpenTelemetry gRPC exporter.

Reviewer: ```suggestion
2. OpenTelemetry Collector fetches metrics from [APXSIX Prometheus plugin](https://apisix.apache.org/docs/apisix/plugins/prometheus/) via Prometheus Receiver and pushes metrics to SkyWalking OAP Server via the OpenCensus gRPC Exporter or OpenTelemetry gRPC exporter.
```

3. The SkyWalking OAP Server parses the expression with [MAL](../../concepts-and-designs/mal.md) to filter/calculate/aggregate and store the results.

### Set up

> 1. Enable APISIX [APXSIX prometheus plugin](https://apisix.apache.org/docs/apisix/plugins/prometheus/) .

Reviewer: ```suggestion
1. Enable APISIX [APXSIX Prometheus plugin](https://apisix.apache.org/docs/apisix/plugins/prometheus/) .
```

2. Set up [OpenTelemetry Collector ](https://opentelemetry.io/docs/collector/getting-started/#docker). For details on Prometheus Receiver in OpenTelemetry Collector, refer to [here](../../../../test/e2e-v2/cases/apisix/otel-collector/otel-collector-config.yaml).
3. Config SkyWalking [OpenTelemetry receiver](opentelemetry-receiver.md).

### MySQL Monitoring
[APXSIX prometheus plugin](https://apisix.apache.org/docs/apisix/plugins/prometheus/) provide multiple dimensions metrics for APISIX server , node , route , etc. 

> Accordingly , Skywalking APISIX monitoring monitor status, resources through APISIX server , node, route aspects . APISIX server is cataloged as a `Layer: APISIX` `Service` in OAP, APISIX node as a `Layer: APISIX` `Instance `, and APISIX route as a `Layer: APISIX` `Endpoint`,  

Reviewer: ```suggestion
Accordingly, Skywalking observes the status, payload, and latency of the APISIX server, which is cataloged as a `LAYER: APISIX` `Service` in the OAP. Meanwhile, the server node would be recognized as `instance, and the route rule would be recognized as `endpoint`.

```

> #### Supported Metrics 

Reviewer: @gxthrj Could you verify the following metrics meanings of APISIX?

3. The SkyWalking OAP Server parses the expression with [MAL](../../concepts-and-designs/mal.md) to filter/calculate/aggregate and store the results.

### Set up
1. Enable APISIX [APISIX Prometheus plugin](https://apisix.apache.org/docs/apisix/plugins/prometheus/) .
2. Set up [OpenTelemetry Collector ](https://opentelemetry.io/docs/collector/getting-started/#docker). For details on Prometheus Receiver in OpenTelemetry Collector, refer to [here](../../../../test/e2e-v2/cases/apisix/otel-collector/otel-collector-config.yaml).
3. Config SkyWalking [OpenTelemetry receiver](opentelemetry-receiver.md).

### MySQL Monitoring
[APISIX prometheus plugin](https://apisix.apache.org/docs/apisix/plugins/prometheus/) provide multiple dimensions metrics for APISIX server , node , route , etc. 
Accordingly, Skywalking observes the status, payload, and latency of the APISIX server, which is cataloged as a `LAYER: APISIX` `Service` in the OAP. Meanwhile, the server node would be recognized as `instance, and the route rule would be recognized as `endpoint`.

#### Supported Metrics 
| Monitoring Panel | Unit | Metric Name |Catalog | Description | Data Source |
|-----|------|-----|-----|-----|-----|

> |HTTP status(per second)  |  | meter_apisix_sv_http_status |Service | The rate of http status trend| APISIX prometheus plugin|

Reviewer: ```suggestion
|HTTP status trend  |  | meter_apisix_sv_http_status |Service | The increment rate of the status of HTTP requests| APISIX Prometheus plugin|
```

Please make case sensitivity correct. Also, `per second` in the name is very confusing. The rate of trend doesn't make sense.

Reviewer: All followings should be updated too.

|HTTP latency(per second) |  | meter_apisix_sv_http_latency |Service | The rate of http latency trend| APISIX prometheus plugin|
|HTTP bandwidth(per second)  | KB | meter_apisix_sv_bandwidth |Service | The rate of http latency trend| APISIX prometheus plugin|
|HTTP status(Not matched route , per second)  |  | meter_apisix_sv_http_status |Service | The rate of http status trend, which request not match any route| APISIX prometheus plugin|
|HTTP latency(Not matched route , per second) |  | meter_apisix_sv_http_latency |Service | The rate of http latency trend, which request not match any route| APISIX prometheus plugin|
|HTTP bandwidth(Not matched route , per second)  | KB | meter_apisix_sv_bandwidth |Service | The rate of http latency trend ,which request not match any route| APISIX prometheus plugin|

> |HTTP connection |  | meter_apisix_sv_http_connections |Service | The avg number of current connection | APISIX prometheus plugin|

Reviewer: ```suggestion
|HTTP connection |  | meter_apisix_sv_http_connections |Service | The avg number of the connections | APISIX Prometheus plugin|
```

> |HTTP request  |  | meter_apisix_http_requests_total |Service | The number of request since APISIX startup | APISIX prometheus plugin|

Reviewer: Why we are going to provide a metric for request counter counted from the startup? This seems pointless.
We should provide a `request traffic load` by using `rate(PT1M)`.

Author: Update  ,  provided  `HTTP Request Trend (per second)`

|Shared dict capacity | MB  | meter_apisix_shared_dict_capacity_bytes |Service | The avg capacity of shared dict capacity | APISIX Prometheus plugin|
|Shared free space| MB  | meter_apisix_shared_dict_free_space_bytes |Service | The avg free space of shared dict capacity | APISIX Prometheus plugin|

> |Etcd index|   | meter_apisix_sv_etcd_indexes |Service | Etcd modify index for APISIX keys | APISIX Prometheus plugin|

Reviewer: `Etcd` -> etbd, ref https://etcd.io/, all lower cases.

Author: I  see , follow official definition

> |Etcd Reachable|   | meter_apisix_sv_etcd_reachable |Service | Etcd Reachable , See [APISIX Prometheus plugin](https://apisix.apache.org/docs/apisix/plugins/prometheus/)| APISIX Prometheus plugin|

Reviewer: Reachable should be a linear graph, because a single value would face 0.5 reachable in some cases, then, what does it mean?

Author: 0 means etcd is  reachable   , otherwise 1 . There only are 0/1  here  . We display latest state .

Author: ```
 apisix_etcd_reachable.downsampling(LATEST).service(['host_name'] , Layer.APISIX)
```
